cql3/column_condition.cc

```diff
@@ -102,170 +93,20 @@ column_condition::column_condition(const column_definition& column, std::optiona
     //
     // We adjust for it by reinterpreting the returned value as a list, since the map
     // representation is not needed here.
-    _column = expr::adjust_for_collection_as_maps(_column);
+    _expr = expr::adjust_for_collection_as_maps(_expr);
 
-    if (op == expr::oper_t::LIKE) {
-        auto literal_term = _value ? expr::as_if<expr::constant>(&*_value) : nullptr;
-        if (literal_term
-                && (type_of(*literal_term) == utf8_type || type_of(*literal_term) == ascii_type)
-                && !literal_term->is_null()) {
-            auto pattern_value = literal_term->type->deserialize(to_bytes(literal_term->view()));
-            auto pattern = value_cast<sstring>(std::move(pattern_value));
-            _matcher = std::make_unique<like_matcher>(bytes_view(reinterpret_cast<const int8_t*>(pattern.data()), pattern.size()));
-        }
-    }
+    _expr = expr::optimize_like(_expr);
 
-    if (op != expr::oper_t::IN) {
-        assert(_in_values.empty());
-    }
+    _expr = update_for_lwt_null_equality_rules(_expr);
```

Comment:
If I understand correctly, these three calls copy _expr three times, even if there is no need (e.g., optimize_like() will usually not do anything). Could these functions be changed to modify the expression in-place instead of always creating a new copy?

Reply:
There's no way to modify in-place. It bothers me too, but of minor importance since it's done at prepare time.

Reply:
If there is no way to modify in-place, maybe there can at least be a way for a function that does nothing (e.g., if there is no LIKE at all) to at least not copy - maybe just move the old expression?

Reply:
Yes, but why? We have to traverse the entire thing, so complexity is unchanged. It's not going to impact performance.

Reply:
Complexity doesn't change but may it slows this step by a large constant factor?
Anyway, I agree this can be optimized later.

Reply:
These checks will make one case faster (when no changes need to be made) and another case faster (when we have to both check and rewrite). I think prepare should favor simplicity.